File: A_Three_Decks.cpp
#include <bits/stdc++.h>
using namespace std;

void solve() {
    int a,b,c;
    cin >> a >> b >> c;  

    int x = a+b+c;

    if(x%3 != 0){
        cout << "NO" <<endl;
        return;
    }
    
    cout << (b > x/3 ? "NO" : "YES") << endl;
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int t;
    cin >> t;
    while (t--) {
        solve();
    }
    return 0;
}

File: A_Three_Decks.py
def solve():
    a, b, c = map(int, input().split())
    x = a + b + c

    if x % 3 != 0:
        print("NO")
        return

    print("NO" if b > x // 3 else "YES")

t = int(input())
for _ in range(t):
    solve()
